### PR TITLE
Problem statement webview

### PR DESCRIPTION
**Pull request by terlan98**
Integrated the new problem statement component inside an `ArtemisWebView`.

### CODE REVIEW

**Comment by terlan98** [comment]
I resolved the merge conflicts, but this PR should not be merged yet, since there's a problem with the new standalone problem statement component. It can't fetch exercises belonging to exams ☹️ 

EDIT: This issue has been fixed. The PR **can** be merged.

File: Themis/ViewModels/Authentication/PreviewAuthenticationViewModel.swift [hunk 1]
```diff
@@ -22,20 +22,24 @@ import UserStore
...
             fatalError("Set the staging user in the environment to use PreviewAuthenticationViewModel.")
         }
+        UserSession.shared.saveUsername(username: username)
         self.username = username
         
         guard let password = stagingPassword else {
             fatalError("Set the staging password in the environment to use PreviewAuthenticationViewModel.")
         }
+        UserSession.shared.savePassword(password: password)
         self.password = password
```
Comment (maximiliansoelch) [inline Themis/ViewModels/Authentication/PreviewAuthenticationViewModel.swift:43]: What has this to do with the problem statement?

File: Themis/Views/Exercises/ProblemStatementView.swift [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    @State private var request: URLRequest
+    @State private var height: CGFloat = .l
+    @State private var isLoading = true
+    
+    init(courseId: Int, exerciseId: Int) {
+        if let url = URL(string: "/courses/\(courseId)/exercises/\(exerciseId)/problem-statement", relativeTo: UserSession.shared.institution?.baseURL) {
+            self._request = State(wrappedValue: URLRequest(url: url))
+        } else {
+            // swiftlint:disable:next force_unwrapping
+            self._request = State(wrappedValue: URLRequest(url: URL(string: "https://google.com")!))
```
Comment (maximiliansoelch) [inline Themis/Views/Exercises/ProblemStatementView.swift]: Can we avoid this?

**Review by maximiliansoelch — CHANGES_REQUESTED** [review]
Can we fix the display of UML Diagrams in the problem statement view of the exercise and a submission? 
Simulator Screen Shot - iPad Pro (12 9-inch) (6th generation) - 2023-06-02 at 12 17 11
Simulator Screen Shot - iPad Pro (12 9-inch) (6th generation) - 2023-06-02 at 12 18 49

Further, can we cache the problem statement so we do not have to load it all the time?

https://github.com/ls1intum/Themis/assets/9954674/04ae672d-9a30-454e-89b9-875f9ba1a076

File: Themis/ViewModels/Authentication/PreviewAuthenticationViewModel.swift (hunk 1 above)
Comment (terlan98) [reply Themis/ViewModels/Authentication/PreviewAuthenticationViewModel.swift:43]: This change was needed to fix the `AuthenticatedPreview`, which is needed to preview the `ProblemStatementView`

File: Themis/Views/Exercises/ProblemStatementView.swift (hunk 2 above)
Comment (terlan98) [reply Themis/Views/Exercises/ProblemStatementView.swift]: Done

**Comment by terlan98** [comment]
The issue on the **first screenshot** was due to the ArtemisWebView not setting it's height correctly when the connection speed was slow. This [ArtemisCore PR](https://github.com/ls1intum/artemis-ios-core-modules/pull/8) will fix it.

---
The issue on the **second screenshot**, where the UML diagram is cut off horizontally, will be fixed with this [Artemis PR](https://github.com/ls1intum/Artemis/pull/6667)

---
**Regarding the caching-related question**: I've looked into several ways of implementing a cache (NSURLProtocol and WKURLSchemeHandler) but unfortunately WKWebView is very restricted and does not allow intercepting requests and responses to cache them. I also tried downloading the webpage content as a file and then loading it from the file for subsequent requests. This approach also did not lead to any significant loading speed improvements. 

So instead of implementing a cache, what I did in my latest commit was to prevent ProblemStatementView from being redrawn when the user interacts with the Picker. The page will still reload if the user hides the sidebar completely and reopens it.

**Review by maximiliansoelch — APPROVED** [review]
Tested locally, Problem statement view works as expected.
Code LGTM 👍

**Pull request merged**